ENH: updates to editor & reviewer onboarding pages (#292)
pyOpenSci/software-peer-review: how-to/onboarding-guide.md

can sometimes be the trickiest part of the review process. However, we have
resources available to help you in that effort!

Below we discuss processes for finding and onboarding new volunteers to our
peer review process.

## About the Editorial Board

The success of our peer review process is dependent upon a well-balanced
editorial board. Our board needs to have combined expertise in:

* A suite of specific science domains that fall within the scope of our peer
  review process
* Technical expertise in Python packaging
* Awareness of the importance of documentation and package usability
* Awareness of the importance of CI / test suites to ensure robust software
  development

We also strive to ensure our editorial team is diverse and comprised of people
from different backgrounds, cultures, genders, and domains.

### Types of Editors

#### New Editors Start as "Guests"

A new editor will be considered a guest editor for the first 3 months of their
tenure and/or until they have completed their first review. Once they have
completed a review, they can be considered a full editor as deemed appropriate
by the software review lead and the current editorial board.

> #### Adhoc Guest Editors

Reviewer: ```suggestion
#### _ad hoc_ Guest Editors
```

> Adhoc editors are editors with specific skill sets that are brought in to lead

Reviewer: ```suggestion
_ad hoc_ editors are editors with specific skill sets that are brought in to lead
```

> a single review. Examples of when there might be a need for an ad-hoc editor

Reviewer: ```suggestion
a single review. Examples of when there might be a need for an _ad hoc_ editor
```

include:

* If there is a conflict of interest between a package submitter and the
  editorial team (e.g., a close colleague of everyone on the team)
* If the editorial board is at capacity handling the current review load
* If a very specialized skill set is needed (in a one-off type of situation)

In this case, you may consider using our internal reviewer sign-up list to see
if someone who signed up to be a reviewer might want to serve as an editor.

### Experience Required to Be an Editor

We prefer that editors have some experience with reviewing software. This
experience could come from a previous review they worked on with pyOpenSci,
rOpenSci, or JOSS.

### Editorial Mentorship

If a potential volunteer does not have prior software editorial experience, we
offer a **mentorship process**. Editor mentorship is where someone with
existing editorial experience mentors the new editor through their first
review(s).

### Recruiting New Editors

Recruiting new editors and maintaining a sufficient and well-balanced
editorial board is the responsibility of the

> [Software Review Lead](https://www.pyopensci.org/handbook/governance/structure.html#software-review-lead)

Reviewer: flagging that it was earlier referred to as peer review lead - I'm assuming these are the same position?

Author: it is. i'll make it consistent locally!

process and track the volume of submissions that we need to support. In the
future, we will find someone with interest in leading peer review for
pyOpenSci.
:::

## Where to Find New Editors

Typically the Editor in Chief will work with the Peer Review Lead to find and
onboard new editors in scientific topical areas where pyOpenSci has existing
gaps.

You might find good candidates to be an editor through:

* Your professional network: you may know people in a specific domain that
  might be a good fit to be an editor for pyOpenSci.
* Our [pyOpenSci list of contributor pool](https://www.pyopensci.org/our-community/index.html#pyopensci-community-contributors) which includes:
  * Package authors and maintainers who have already submitted a package for
    review to pyOpenSci
  * Reviewers who have already led reviews for pyOpenSci
  * Past guest editors
* Colleagues that you know who have reviewed for the Journal of Open Source
  Software (JOSS) or rOpenSci who have Python expertise
* Community members in the pyOpenSci Slack (please post a call in our
  `#software-review` channel)

When all of the above fails to return a good new editor candidate, you can find

> support from our pyOpenSci community manager who will post an open call on our

Reviewer: ```suggestion
support from our pyOpenSci Community Manager who will post an open call on our
```

Reviewer: while it's _technically_ correct to keep job titles all lower case, it's culturally//stylistically preferred to have them capitalized

If you have tried all of the above and still aren't able to find a new
candidate, then the pyOpenSci Community Manager can assist you by:

* Posting on our social media channels about the position.
* Writing a [call for editors blog post](https://www.pyopensci.org/blog/pyos-call-for-editors-may-2024.html) about our current editorial needs.

:::{note}
[See an example here of a post that seeks to recruit editors](https://fosstodon.org/@pyOpenSci/112497485980274296), a blog post, and call-outs in both our monthly and weekly newsletters.
:::

To get support from the Community Manager in recruiting editors (or reviewers) follow the process below:

* Post in our public `#software-review` slack channel pinging the community manager (`@username`). In your post, include the following details:

  * That you’re looking for editors.
  * Any specific domains or technical skills you need an editor to have, or if it’s a general call for editors.
  * Any specific packages that need an editor (include links to a package submission that they would lead if possible).

:::{tip}
It's helpful to post in our `#software-review` channel rather than
an individual Direct Message (DM) because:

> * It allows other editors to see the type of expertise that you are requesting and that we offer this support from our community manager.

Reviewer: ```suggestion
* It allows other editors to see the type of expertise that you are requesting and that we offer this support from our Community Manager.
```

* It allows other community members who may know of potential editors to see the request and potentially respond.
:::

> Our Community Manager interacts with the broader community and partner communities and as such often interacts with people who may be interested in joining our pyOpenSci review team. The community manager may from time to time meet community members who might make great editors or reviewers. In those cases they will share that information with the current EiC and peer review lead.

Reviewer: ```suggestion
Our Community Manager interacts with the broader community and partner communities and as such often interacts with people who may be interested in joining our pyOpenSci review team. The Community Manager may from time to time meet community members who might make great editors or reviewers. In those cases they will share that information with the current Editor in Chief and Peer Review Lead.
```

:::

:::{note}

It is important that a new reviewer or a new editor fill out the appropriate form prior to onboarding.

* [reviewer signup form](https://docs.google.com/forms/d/e/1FAIpQLSeVf-L_1-jYeO84OvEE8UemEoCmIiD5ddP_aO8S90vb7srADQ/viewform)
* [editor signup form](https://docs.google.com/forms/d/17NW0P_h8gpmzf7Fr0cAd8aEbIUWPljPfg4d7Rjs1UIE/edit)

:::

## On-boarding a new editor

The Editor in Chief, working closely with the Peer Review Lead, is responsible for inviting and onboarding a new editor to our peer review process.

When the EIC has identified an editor who is not currently part of the pyOpenSci community, they should:

* Extend a Slack invitation to the individual (or ask our Community Manager to do so)
* Welcome the individual in Slack, and provide them with access to the `#private-editorial-team` and any other channels as needed.

:::{note}
The Community Manager will ensure that any new member that joins our Slack workspace is welcomed and set up with anything they need in our pyOpenSci Slack workspace.
:::

## Process for inviting a new editor

* Editorial board candidates most often start as guest editors.

> * After 3 months or their first review (which ever comes first), the peer review lead working with the EiC will assess how the

Reviewer: ```suggestion
* After 3 months or their first review (which ever comes first), the Peer Review Lead working with the EiC will assess how the
```

- name: FirstName LastName
  github_username: ghuser
  github_image_id: 1234566
  title:
    - Editor # Title to be listed under their name
  # ..... #
  editorial_board: true # Be sure editorial_board is set to true
  emeritus_editor: false # Emeritus is initially set to false
  # ..... #
  contributor_type:
    - editor # This field will be automatically updated after their first review
    - guidebook-contrib
    - reviewer
```

## Off-boarding an editor

When it is time for an editor to step down, do the following:

* Thank them for their work!
* Remove them from the editors-only Slack channel and the editors GitHub team.
* Move them to emeritus-editor on the [pyOpenSci website](https://github.com/pyOpenSci/pyopensci.github.io/blob/main/_data/contributors.yml) as follows:

```yaml
- name: FirstName LastName
  github_username: ghuser
  github_image_id: 1234566
  # ..... more here ... #
  editorial_board: false # Be sure editorial_board is set to FALSE
  emeritus_editor: true # Emeritus is now true if they've served as an editor
  # .. #
  contributor_type:
    - editor # This field will be automatically updated after their first review
    - guidebook-contrib
    - reviewer

> ```

Author: TODO: add - onboarding new reviewers section!

  * For ad-hoc guest editors: invite the editor to the `software-submission` repository
    with "maintain" permissions.
  * For new editors (not ad-hoc): invite editors to the pyOpenSci GitHub organization
    as a member of the [`editorial-board`](https://github.com/orgs/pyOpenSci/teams/editorial-board) team. This will give them appropriate permissions and allow them to get team-specific notifications.

* Add the new editor to the pyOpenSci Slack workspace and specifically the `private-editorial` channel.

* Post a welcome message for the new editor in the editor channel, pinging all editors.
* Update the [website contributors.yml](https://github.com/pyOpenSci/pyopensci.github.io/blob/main/_data/contributors.yml) file with the name of the new editor.

:::{note}
We have a bi-weekly cron job that parses through existing issues and grabs the names of editors, reviewers and authors. However, if you want the editors name to be listed on the website prior to a review beginning and/or sooner than the cron job might pick up their name, then we suggest that you add their name, and GitHub username and title to the contributors.yml file through a pull request.

> You do not need to fill out all of the elements of the YAML file - only the name, github user field and the `editorial_board: true` key:value pair.

Reviewer: ```suggestion
You do not need to fill out all of the elements of the YAML file - only the name, GitHub user field and the `editorial_board: true` key:value pair.
```
